libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/Store.kt

```diff
@@ -91,11 +94,10 @@ open class Store {
         return false
     }
 
-    private fun reduce(action: Action) {
-        reducers.forEach { reducer ->
-            _state.value = reducer.process(_state.value, action)
+    private fun reduce(state: State, action: Action): State =
```

Comment:
Should the same thing be done for the `intercept` function?

Curious if there is a race condition with the `_state.value`

``` kotlin
    private fun intercept(state: State, action: Action): Boolean {
        middlewares.forEach {
            if (it.onDispatch(state, action)) return true
        }
        return false
    }
```

Reply:
I wouldn't worry about this. There might be situations where we would like to register a middleware that consumes an action and dispatches a different action instead.